### PR TITLE
Update docs contribution guide for two-repo setup

### LINKED ISSUE #3: Update contributor docs
## 📚 Documentation

We need to make sure that contributor docs are brought up to date given our current two repo setup:

https://napari.org/dev/developers/documentation/index.html

### PR DESCRIPTION
**Pull request by melissawm**
# Description
Updates the docs contribution guide to match the new documentation setup.

Also some light copyediting of the content.

## Type of change

- [ ] Bug-fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# References
Closes #3 

# How has this been tested?

- [ ] example: the test suite for my feature covers cases x, y, and z
- [ ] example: all tests pass with my change

## Final checklist:
- [ ] My PR is the minimum possible work for the desired functionality
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] If I included new strings, I have used `trans.` to make them localizable.
      For more information see our [translations guide](https://napari.org/docs/dev/guides/translations.html).

### CODE REVIEW

File: docs/developers/documentation/index.md
Comment (melissawm) [inline docs/developers/documentation/index.md]: Not the best solution, but the Makefile needs to be updated in a follow-up, and this can be corrected accordingly.

**Review by jni — APPROVED** [review]
I'm happy to merge as-is, but wanted to make a couple of notes — could be fixed here or in follow-up PRs:

- I wasn't sure how to update the examples dir in the sphinx configuration, so currently the docs build actions [manually copy](https://github.com/napari/docs/blob/42313f68bb6cbbcffcebf3029bd16a60faeea6ab/.github/workflows/deploy_docs.yml#L34-L36) the examples directory into this repo. Since this PR updates the sphinx config, perhaps it's worth updating the action as well? Or it might be worth splitting out that configuration update into a separate PR?
- I think many docs changes will be fine to make in the GH interface or a plaintext editor, but a *really nice* feature of jupytext is that it allows you to run the .md files from a Jupyter Lab or Jupyter Notebook interface. This is one of the main reasons why we went with myst-md rather than plain notebooks. I think we should explicitly encourage this way of writing docs, including a final clear+run-all, as the preferred mode of contribution. Again, this could be done in a follow-up.

**Comment by jni** [comment]
> Or it might be worth splitting out that configuration update into a separate PR?

**Update**: Just saw that this was done in #6. 🙏 But the deploy action and the action on the main repo still need to be updated.

File: README.md
Comment (lucyleeow) [inline README.md]: Just a question, is the autogenerated API doc from the docstrings different from what is mentioned below:

> Meanwhile, changes to docstrings or to the
[examples gallery](https://napari.org/gallery) should be made to the
[napari/napari](https://github.com/napari/napari) repository.

File: docs/developers/documentation/index.md
Comment (lucyleeow) [inline docs/developers/documentation/index.md]: Could we add a link?
```suggestion
and clone the [`napari/docs`](https://github.com/napari/docs) repo to your machine, and navigate to the resulting
```

File: docs/developers/documentation/index.md
Comment (lucyleeow) [inline docs/developers/documentation/index.md]: maybe?
```suggestion
  require a clean conda environment but with a development installation of
```

File: docs/developers/documentation/index.md
Comment (lucyleeow) [inline docs/developers/documentation/index.md:55]: Just a question, is jupyter notebook included when doing the dev installation?

File: docs/developers/documentation/index.md
Comment (lucyleeow) [inline docs/developers/documentation/index.md]:
```suggestion
Alternatively, when you submit your pull request, napari docs continuous integration
```

File: docs/developers/documentation/index.md
Comment (lucyleeow) [inline docs/developers/documentation/index.md:266]: Outside scope of this PR, but with your docs build comment addition, I am wondering if this should be amended or removed.

**Review by lucyleeow — COMMENTED** [review]
Just a quick look, thanks for doing this, it looks great!

**Comment by melissawm** [comment]
Thanks for the reviews folks - I will get to it as soon as possible.

File: docs/conf.py
Comment (psobolewskiPhD) [inline docs/conf.py]: I think you need to make sure to merge main here, else building docs on napari/napari PR will break again.

File: README.md
Comment (psobolewskiPhD) [reply README.md]: I think you're right, examples and docstrings live in napari/napari. This repo is just the authored stuff like tutorials, guides, etc on napari.org

File: docs/developers/documentation/index.md
Comment (psobolewskiPhD) [inline docs/developers/documentation/index.md:40]: Is this strictly true? I was thinking that with the docs being in a separate repo people could use GitHub.dev or the normal "pencil" tool on GitHub to edit docs, because they are all .md files.
If true, this would massively lower the "activation energy required".

**Review by psobolewskiPhD — APPROVED** [review]

File: docs/conf.py
Comment (Czaki) [reply docs/conf.py]: I strongly prefer to make it more configurable. People could have different structures of their repositories, and it should be possible to build docs. 
Maybe some environment variable with the path to the napari repository (with default to the currently proposed path).

File: docs/developers/documentation/index.md
Comment (lucyleeow) [reply docs/developers/documentation/index.md:40]: Fair point, in the docs meeting (https://hackmd.io/10DKrIMhREmk9sa_QRirlg) we did talk about wanting to highlight the “upload via web interface” workflow - if you are editing or adding just one file, this can be done via the github web interface then just use the CI to view the built documentation.

I think you can quite easily upload several files via web interface: https://docs.github.com/en/repositories/working-with-files/managing-files/adding-a-file-to-a-repository
However it is not as easy to edit multiple files and open one PR: https://stackoverflow.com/questions/17815895/can-i-edit-two-files-then-make-one-commit-using-github-web-based-editor - I am not sure this is necessarily easier than doing it via git...

I think we could update the docs to say that if you are adding file(s) or just amending one file, this can be done via web interface without installing git or making a new environment. This may be outside the scope of this PR though and could be done after this is merged.

File: docs/conf.py
Comment (lucyleeow) [reply docs/conf.py]: An environment variable sounds like a good option but I am not sure how it would work? (maybe just something I don't know about).

Not sure if this is helpful but Sphinx allows overriding conf.py values from the command line (see last note [here](https://sphinx-gallery.github.io/stable/configuration.html#parsing-and-executing-examples-via-matching-patterns)) - thus the value in the `conf.py` file is effectively the default and the command in the makefile could be made configurable somehow....? E.g., to set `examples_dirs` you can do:
 
`sphinx-build -D sphinx_gallery_conf.examples_dirs=<my_local_path>`

File: docs/developers/documentation/index.md
Comment (lucyleeow) [inline docs/developers/documentation/index.md]:
```suggestion
The [**Examples gallery**](../../gallery) sources are in the [main `napari/napari` repository](https://github.com/napari/napari/tree/main/examples)
and show code examples of how to use napari.
```

File: README.md
Comment (melissawm) [reply README.md]: That is right, I'll fix it.

File: docs/developers/documentation/index.md
Comment (melissawm) [reply docs/developers/documentation/index.md:55]: It is when installing the docs dependencies - added this to the text.

File: docs/developers/documentation/index.md
Comment (melissawm) [reply docs/developers/documentation/index.md:266]: Let me do this in a follow up, after making sure we have a stable action for this. Thanks!

File: docs/developers/documentation/index.md
Comment (melissawm) [reply docs/developers/documentation/index.md:40]: Fair! I'll open an issue for this.

Comment (melissawm) [reply docs/developers/documentation/index.md:40]: Here you go: #34

**Comment by psobolewskiPhD** [comment]
Build is failing because of the change noted here:
https://github.com/napari/docs/pull/5#discussion_r997488663
Need:
```
sphinx_gallery_conf = {
    'examples_dirs': '../examples',
```
due to https://github.com/napari/docs/pull/8

File: docs/conf.py
Comment (melissawm) [reply docs/conf.py]: Definitely better - thanks @lucyleeow , I'll include this suggestion in this PR.

**Comment by melissawm** [comment]
Ok - I think this is a reasonable solution. If you clone both repos next to each other, nothing has changed and you can simply use `make docs`. But if you want to specify the examples gallery location, you can use

```
$ make docs GALLERY_PATH=<your path>
```

The github workflow has been updated accordingly as well.

**Comment by melissawm** [comment]
Hi folks, any feedback on the proposed solution here? @psobolewskiPhD @jni

File: docs/developers/documentation/index.md
Comment (psobolewskiPhD) [inline docs/developers/documentation/index.md:238]: This should probably be just `docs` right? I mean I also renamed my fork, but for the general case?

**Review by psobolewskiPhD — APPROVED** [review]
Just to be clear, if I have this:
```
~/dev/napari/napari
~/dev/docs/docs
```
then everything works without the env var, but if I have some other combo, then need the env var?

File: docs/developers/documentation/index.md
Comment (melissawm) [reply docs/developers/documentation/index.md:238]: Well, since in the above instructions I actually cloned it into napari-docs, I've kept the same convention here. I can replace this with just docs everywhere, or make sure folks know they can rename  and use something like <docs repo clone folder>

**Comment by melissawm** [comment]
> Just to be clear, if I have this:
> 
> ```
> ~/dev/napari/napari
> ~/dev/docs/docs
> ```
> 
> then everything works without the env var, but if I have some other combo, then need the env var?

Yes - the current relevant snippet from the Makefile is

```
ifeq ($(GALLERY_PATH),)
GALLERY_PATH := ../../napari/examples
endif
```

So if the repos are cloned "next to each other", all should be good.

File: docs/developers/documentation/index.md
Comment (psobolewskiPhD) [reply docs/developers/documentation/index.md:238]: Derp, yeah I see it now. Sorry! I agree, let's leave it to be consistent.
I personally hate the `docs` name, but it is what it is. 
Still as a relative newbie, and putting on my end-user-hat, I think I would appreciate the explicit clarification that the `napari/docs` repo was cloned as `napari-docs` with a clear descriptive phrase. I think potential `docs` fixer-upers may not be git or github gurus so the extra explanation could be worth it.
Just something like:
Note: we've renamed the repo to napari-docs to avoid confusion    ?

Comment (melissawm) [reply docs/developers/documentation/index.md:238]: Yeah, makes sense. I personally have this setup, where I clone it into napari-docs. I have a similar situation with github.com/numpy/archive and github.com/scipy/archive so I've been using this convention already, but I'm happy to make it clearer!

File: docs/developers/documentation/index.md
Comment (psobolewskiPhD) [inline docs/developers/documentation/index.md:32]:
```suggestion

````{note}
To reduce confusion and possible conflicts, the `docs` fork is being cloned into a local repository named `napari-docs`. Alternately, you could also rename the repository when forking `napari/docs`.
```

Comment (melissawm) [reply docs/developers/documentation/index.md:32]: Included- thanks!

**Comment by jni** [comment]
❤️ 🚀

**Pull request merged**
